workbench-core/datasets/src/dataSetService.ts

```diff
@@ -313,6 +344,41 @@ export class DataSetService {
     return await this._dbProvider.listStorageLocations();
   }
 
+  //
+  // DataSet Permissions
+  //
+
+  /**
+   * Add AccessPermissions to a DataSet.
+   *
+   * @param params - a {@link AddRemoveAccessPermissionRequest} object indicating the datasetId and the
+   *                 requested permissions.
+   * @returns a {@link PermissionsResponse} object containing the permissions added.
+   */
+  public async addDataSetAccessPermissions(
+    params: AddRemoveAccessPermissionRequest
+  ): Promise<PermissionsResponse> {
+    const metadata: Metadata = {
+      actor: { id: params.authenticatedUserId, roles: params.roles },
+      action: this.addDataSetAccessPermissions.name,
+      source: {
+        serviceName: DataSetService.name
+      },
+      requestBody: params
+    };
+
+    try {
+      // this will throw if the dataset is not found.
+      await this.getDataSet(params.dataSetId);
+      const response: PermissionsResponse = await this._authzPlugin.addAccessPermission(params);
+      await this._audit.write(metadata, response);
```

Comment:
status code missing here

Reply:
see previous comment about status codes being optional and confusing.